src/site/content/en/blog/mediastreamtrack-insertable-media-processing/index.md

```diff
@@ -0,0 +1,257 @@
+  `"video"`, the stream accepts `VideoFrame` objects and fails with any other type. When a frame
+  is written to `writable`, the frame's `close()` method is automatically invoked, so that its
+  internal resources are no longer accessible from JavaScript.
+- `readableControl`: A `ReadableStream` that allows reading control signals sent from any sinks
+  connected to the `MediaStreamTrackGenerator`. Control signals are objects of type
+  `MediaStreamTrackSignal`.
+
+In the `MediaStream` model, apart from media, which flows from sources to sinks, there are also
+control signals that flow in the opposite direction (i.e., from sinks to sources via the track). A
+`MediaStreamTrackProcessor` is a sink and it allows sending control signals to its track and source
+via its `writableControl` property. A `MediaStreamTrackGenerator` is a track for which a custom source
+can be implemented by writing media frames to its `writable` field. Such a source can receive
+control signals sent by sinks via its `readableControl` property.
+
+### Bringing it all together
+
+The core idea is to create a processing chain as follows:
+
+Platform Track → Processor → Transform → Generator → Platform Sinks
+
+For a barcode scanner application, this chain would look as in the code sample below.
+
+```js
+const stream = await getUserMedia({ video: true });
+const videoTrack = stream.getVideoTracks()[0];
+
+const trackProcessor = new MediaStreamTrackProcessor(videoTrack);
+const trackGenerator = new MediaStreamTrackGenerator();
+
+const transformer = new TransformStream({
+  async transform(videoFrame, controller) {
+    const barcodes = await detectBarcodes(videoFrame);
+    const newFrame = highlightBarcodes(videoFrame, barcodes);
+    videoFrame.close();
+    controller.enqueue(newFrame);
+  },
+});
+
+trackProcessor.readable.pipeThrough(transformer).pipeTo(trackGenerator.writable);
+
+trackGenerator.readableControl.pipeTo(trackProcessor.writableControl);
+```
+
+{% Aside %} This article barely scratches the surface of what is possible and going into the details
+is way beyond the scope of this publication. For more examples, see the advanced
+[video processing demo](https://webrtc.github.io/samples/src/content/insertable-streams/video-processing/)
+and the advanced
+[audio processing demo](https://webrtc.github.io/samples/src/content/insertable-streams/audio-processing/)
+respectively. You can find the source code for both demos
+[on GitHub](https://github.com/webrtc/samples/tree/gh-pages/src/content/insertable-streams).
+{% endAside %}
+
+## Demo
+
+You can see the [QR code scanner demo](https://mediastreamtrack.glitch.me/) from the section above
+in action on a desktop or mobile browser. Hold a QR code in front of the camera and the app will
+detect it and highlight it. You can see the application's source code
+[on Glitch](https://glitch.com/edit/#!/mediastreamtrack?path=index.html%3A21%3A50).
```

Comment:
One comment on the linked Glitch (sorry, if there's a way to comment there, I don't see it): You're setting up the canvas state in the loadedmetadata event handler on video; however, that video element is playing the output video, not the input video, so that doesn't make much sense. It seems better to set up the canvas just after obtaining videoTrack by looking at videoTrack.getSettings() to get the width and height.

Maybe also move the loadedmetadata handler to the line after setting srcObject.

Reply:
Demo fixed accordingly.